SchoolBase.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace School
+{
+    public abstract class SchoolBase : Sorting
+    {
+        public List<Human> _humans = new List<Human>();
+
+        protected string _name;
+
+        public void AddMember(Human human)
+        {
+            _humans.Add(human);
+        }
+
+        public void ShowMembers()
+        {
+            var i = 1;
+            foreach (var human in _humans)
+            {
+                Console.WriteLine(i++ + ". " + human.Name + " ID: " + human.ID);
+            }
+        }
+        
+        public string Name
+        {
+            get { return _name; }
+            set { _name = value; }
+        }
+
+        private Human FindMember(int ID)
```

Comment:
This goes through the whole _humans list even though it doesn't have to. It should look for the first one that matches the condition and return it straight away. You won't need the temp then (and it shouldn't have been initialised in the first place anyway)